File: 4C - Registration System.py
#4076779   Jul 15, 2013 8:58:49 PM	fuwutu	 4C - Registration System	 Python 3	Accepted	 1828 ms	 600 KB
n = int(input())
d = {}
for i in range(0, n):
    s = input()
    if s in d:
        print(s+str(d[s]))
        d[s] += 1
    else:
        print("OK")
        d[s] = 1

File: 4C - Registration System.cpp
//4076800   Jul 15, 2013 9:04:58 PM	fuwutu	 4C - Registration System	 GNU C++0x	Accepted	 578 ms	 900 KB
#include <iostream>
#include <string>
#include <map>

using namespace std;

int main()
{
    int n;
    cin >> n;

    map<string, int> database;
    string s;
    while (n--)
    {
        cin >> s;
        if (database.count(s) == 0)
        {
            cout << "OK" << endl;
            database[s] = 1;
        }
        else
        {
            cout << s << database[s] << endl;
            database[s] += 1;
        }
    }
    return 0;
}
